tests/cmd/stability/main.go

```diff
@@ -155,58 +169,49 @@ func run() {
-			cluster.TiKVPreStartScript = strconv.Quote("exit 1")
-			cluster.PDPreStartScript = strconv.Quote("exit 1")
-			oa.UpgradeTidbClusterOrDie(cluster)
-			time.Sleep(30 * time.Second)
-			oa.CheckTidbClustersAvailableOrDie([]*tests.TidbClusterConfig{cluster})
-			// rollback conf
-			cluster.PDPreStartScript = strconv.Quote("")
-			cluster.TiKVPreStartScript = strconv.Quote("")
-			cluster.TiDBPreStartScript = strconv.Quote("")
-			oa.UpgradeTidbClusterOrDie(cluster)
-			// wait upgrade complete
-			oa.CheckUpgradeCompleteOrDie(cluster)
-			oa.CheckTidbClusterStatusOrDie(cluster)
-
-			cluster.UpdatePdMaxReplicas(cfg.PDMaxReplicas).
-				UpdateTiKVGrpcConcurrency(cfg.TiKVGrpcConcurrency).
-				UpdateTiDBTokenLimit(cfg.TiDBTokenLimit)
-			oa.UpgradeTidbClusterOrDie(cluster)
-			// wait upgrade complete
-			oa.CheckUpgradeOrDie(ctx, cluster)
-			oa.CheckTidbClusterStatusOrDie(cluster)
+			cluster.Clustrer.Spec.PD.Replicas = int32(cfg.PDMaxReplicas)
+			grpcConcurrency := uint(cfg.TiKVGrpcConcurrency)
+			cluster.Clustrer.Spec.TiKV.Config.Server.GrpcConcurrency = &grpcConcurrency
+			tokenLimit := uint(cfg.TiDBTokenLimit)
+			cluster.Clustrer.Spec.TiDB.Config.TokenLimit = &tokenLimit
+			crdUil.UpdateTidbClusterOrDie(cluster.Clustrer)
+			crdUil.WaitTidbClusterReadyOrDie(cluster.Clustrer, 60*time.Minute)
 		}
-		cancel()
 		oa.CleanWebHookAndServiceOrDie(ocfg.WebhookConfigName)
 		klog.Infof("clusters configurations updated in checked")
 
 		for _, cluster := range clusters {
-			oa.CheckDisasterToleranceOrDie(cluster)
+			crdUil.CheckDisasterToleranceOrDie(cluster.Clustrer)
 		}
 		klog.Infof("clusters DisasterTolerance checked")
 
-		// backup and restore
-		for i := range backupTargets {
-			oa.DeployTidbClusterOrDie(backupTargets[i].TargetCluster)
-			addDeployedClusterFn(backupTargets[i].TargetCluster)
-			oa.CheckTidbClusterStatusOrDie(backupTargets[i].TargetCluster)
-		}
-		oa.BackupAndRestoreToMultipleClustersOrDie(clusters[0], backupTargets)
-		klog.Infof("clusters backup and restore checked")
+		// TODO: Use Backup and Restore CRD instead
+		//backup and restore
+		//for i := range backupTargets {
+		//	tc := backupTargets[i].TargetCluster.Clustrer
+		//	crdUil.CreateTidbClusterOrDie(tc)
+		//	addDeployedClusterFn(backupTargets[i].TargetCluster)
+		//	secret := buildSecret(backupTargets[i].TargetCluster)
+		//	crdUil.CreateSecretOrDie(secret)
+		//	crdUil.CheckDisasterToleranceOrDie(tc)
+		//}
+		//clusterZero := clusters[0].Clustrer
+		//clusterZero = crdUil.GetTidbClusterOrDie(clusterZero.Name, clusterZero.Namespace)
+		//clusterZero = fixture.AddPumpForTidbCluster(clusterZero)
+		//crdUil.UpdateTidbClusterOrDie(clusterZero)
+		//crdUil.WaitTidbClusterReadyOrDie(clusterZero, 30*time.Minute)
+		//oa.BackupAndRestoreToMultipleClustersOrDie(clusters[0], backupTargets)
+		//klog.Infof("clusters backup and restore checked")
+		//slack.NotifyAndCompletedf("stability test: clusters backup and restore checked")
```

Comment:
This case does not involve destructive actions. I think we can delete them and migrate them to ginkgo e2e suite (if no corresponding case exists).

Reply:
updated